coderd/database/queries/activitybump.sql

```diff
@@ -0,0 +1,42 @@
+-- We bump by the original TTL to prevent counter-intuitive behavior
+-- as the TTL wraps. For example, if I set the TTL to 12 hours, sign off
+-- work at midnight, come back at 10am, I would want another full day
+-- of uptime. In the prior implementation, the workspace would enter
+-- a state of always expiring 1 hour in the future.
+-- name: ActivityBumpWorkspace :exec
+WITH latest AS (
+	SELECT
+		workspace_builds.id::uuid AS build_id,
+		workspace_builds.deadline::timestamp AS build_deadline,
+		workspace_builds.max_deadline::timestamp AS build_max_deadline,
+		workspace_builds.transition AS build_transition,
+		provisioner_jobs.completed_at::timestamp AS job_completed_at,
+		(workspaces.ttl / 1000 / 1000 / 1000 || ' seconds')::interval AS ttl_interval,
+		(NOW() AT TIME ZONE 'UTC')::timestamp as now_utc
+	FROM workspace_builds
+	JOIN provisioner_jobs
+		ON provisioner_jobs.id = workspace_builds.job_id
+	JOIN workspaces
+		ON workspaces.id = workspace_builds.workspace_id
+	WHERE workspace_builds.workspace_id = @workspace_id::uuid
+	ORDER BY workspace_builds.build_number DESC
+	LIMIT 1
+)
+UPDATE
+	workspace_builds wb
+SET
+	updated_at = NOW(),
+	deadline = CASE
+		WHEN l.build_max_deadline = '0001-01-01 00:00:00'
+		THEN l.now_utc + l.ttl_interval
```

Comment:
Since we're removing timezone information in the CTE, this will interpret the time in whatever timezone PostgreSQL is configured for, even though it's supposed to represent UTC, this will skew the clock for non-UTC DBs.

I would recommend simply using `NOW()` here and removing the `now_utc` column entirely from the CTE.

Reply:
(Side note: PostgreSQL doesn't store the given timezone in the DB column, values are always stored in the configured PG timezone, but when you insert data in another timezone it will be automatically converted for you, and when a timestamp is inserted without timezone, it will be interpreted as the configured timezone.)

Reply:
@Emyrk nah, not AFAIK. I guess it might be more correct to think about it as storing a unix timestamp, only the representation when you look at it changes.

Reply:
Updated!